src/server/src/server_main.cpp

```diff
@@ -0,0 +1,64 @@
+//
+// Created by kevin on 6/25/22.
+//
+#include <ros/ros.h>
+#include <client/current_time.h>
+#include <client/login.h>
+#include <vector>
+
+#include <iostream>
+ros::ServiceServer service_server;
+std::vector<ros::Subscriber> server_subscribers;
+
+void time_callback(const client::current_time::ConstPtr &msg) {
+    ROS_INFO("Topic received from : %s , Whose time is %d/%d/%d-%d:%d:%d ."
+            , (msg->name).c_str()
+            , msg->year
+            , msg->month
+            , msg->day
+            , msg->hour
+            , msg->minute
+            , msg->second
+    ); //Output
+}
+
+bool login_handle(client::login::Request &req, client::login::Response &ack) {
+    if(req.req_code == 1) {
+        ROS_INFO("A client: %s has logged in!", req.node_name.c_str());
+        ack.ack_code = 10;
+        ros::NodeHandle server_logging_handle;
+        server_subscribers.push_back(
+                        server_logging_handle.subscribe(
+                                req.node_name, 1, time_callback));
+    }else{
+        ROS_INFO("A client: %s has logged out!", req.node_name.c_str());
+        for(auto &each_subscriber : server_subscribers){
+            std::string assist_sig = "/";
+            if(assist_sig + req.node_name.c_str() == each_subscriber.getTopic()){
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]